python_analytics/event_encoder.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import absolute_import, unicode_literals
+
+from abc import ABCMeta, abstractmethod
+from weakref import WeakKeyDictionary
+
+from six import add_metaclass, text_type
+
+
+NoValue = object()
+
+
+@add_metaclass(ABCMeta)
+class _TrackedAttribute(object):
```

Comment:
There are limits on some field sizes that I don't handle now. (e.g. custom dimension has max size of 150 bytes)